### PR TITLE
samples: fix wording for preconditions to use correct operation inste…

### LINKED ISSUE #2129: samples: audit precondition comments
Please check the samples that have preconditions for:
- comments that reference "upload" instead of the current operation being performed with the precondition
- comments that mention "generation" instead of "metageneration", where applicable

See https://github.com/GoogleCloudPlatform/golang-samples/pull/2813 for reference on comparable changes made in Go.

- The delete file sample has this mistake, please fix and audit other samples.

### PR DESCRIPTION
**Pull request by ddelgrosso1**
…ad of upload

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/nodejs-storage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #2129  🦕

### CODE REVIEW

File: samples/setEventBasedHold.js [hunk 1]
```diff
@@ -43,7 +43,7 @@ function main(
 
   async function setEventBasedHold() {
     // Optional: set a generation-match precondition to avoid potential race
-    // conditions and data corruptions. The request to upload is aborted if the
+    // conditions and data corruptions. The request to set metadata is aborted if the
     // object's generation number does not match your precondition.
     const options = {
       ifGenerationMatch: generationMatchPrecondition,
```
Comment (BrennaEpp) [inline samples/setEventBasedHold.js:49]: Shouldn't object metadata updates/patches be conditionally idempotent on the metageneration, not the generation, or am I missing something here/for node?

Comment (ddelgrosso1) [reply samples/setEventBasedHold.js:49]: You are right, let me clean this up and open a separate PR.

**Comment by ddelgrosso1** [comment]
This should be merged after: https://github.com/googleapis/nodejs-storage/pull/2131

**Review by danielbankhead — APPROVED** [review]
Both PRs LGTM

**Review by shaffeeullah — APPROVED** [review]
oops. thanks for fixing this, Denis!!

**Pull request merged**